server/src/main/java/org/elasticsearch/action/admin/cluster/node/tasks/cancel/TransportCancelTasksAction.java

```diff
@@ -276,4 +279,23 @@ public void messageReceived(final BanParentTaskRequest request, final TransportC
         }
     }
 
+    void cancelOrphanedTasks(Set<CancellableTask> orphanedTasks) {
```

Comment:
I think we should move this code (`cancelTaskAndDescendants`) in the TaskManager and expose a simple `TaskManager#cancel` method that this action can call. That would remove the need to have public methods in  the `TaskManager` around bans and orphaned tasks. They should be private and solely used by the TaskManager to cancel a task internally ?

Reply:
Thanks for pushing this 👍 . I've moved the cancellation to TaskManager in https://github.com/elastic/elasticsearch/pull/56620/commits/bd2950455daec1e115a0cbbdafb986c51bc77585.